File: backend/src/main/java/com/NBE_4_5_2/Team5/global/exception/ServiceException.java
package com.NBE_4_5_2.Team5.global.exception;

import com.NBE_4_5_2.Team5.global.dto.RsData;

public class ServiceException extends RuntimeException {
    private final RsData<?> rsData;

    public ServiceException(String code, String message) {
        super(message);
        rsData = new RsData<>(code, message);
    }

    public String getCode() {
        return rsData.getCode();
    }

    public String getMsg() {
        return rsData.getMessage();
    }

    public int getStatusCode() {
        return rsData.getStatusCode();
    }
}

File: backend/src/main/java/com/NBE_4_5_2/Team5/global/exception/ServiceException.kt
package com.NBE_4_5_2.Team5.global.exception

import com.NBE_4_5_2.Team5.global.dto.Empty
import com.NBE_4_5_2.Team5.global.response.RsData

open class ServiceException(
    private val _code: String,
    override val message: String
): RuntimeException(message) {

    private var rsData: RsData<Empty> = RsData(_code, message)

    val code: String
        get() = rsData.code

    val msg: String
        get() = rsData.message

    val statusCode: Int
        get() = rsData.statusCode

}
